src/osd/PG.cc

```diff
@@ -2646,41 +2644,39 @@ void PG::_update_calc_stats()
       //   - not in up  Compute misplaced objects excluding num_missing
       // backfill       Compute total objects already backfilled
       if (!is_backfill_targets(p)) {
-        unsigned osd_missing;
+        unsigned osd_missing, osd_objects;
         // primary handling
         if (p == pg_whoami) {
           osd_missing = pg_log.get_missing().num_missing();
           info.stats.stats.sum.num_objects_missing_on_primary =
               osd_missing;
-          object_copies += num_objects; // My local (primary) count
         } else {
           assert(peer_missing.count(p));
           osd_missing = peer_missing[p].num_missing();
-          // During recovery peer_info[p].stats.stats.sum.num_objects
-          // doesn't reflect the object count, but num_missing
-          // does reflect the missing and decreases as recovery progresses
-          // so use primary's object count.
-          object_copies += num_objects;
         }
-        missing += osd_missing;
+
+        // Safety check so we don't go negative anywhere
```

Comment:
Hmm, can this actually happen? Do we want to stop processing if they have a bunch of objects which the primary doesn't? It seems like that could get us into the opposite problem — we undercount recovery we need to do, instead of overcounting it.